app/src/main/java/nu/yona/app/api/manager/impl/ActivityManagerImpl.java

```diff
@@ -1148,19 +1141,19 @@ public YonaBuddy findYonaBuddy(Href yonaBuddy)
 	private YonaGoal findYonaBuddyGoal(Href goalHref)
 	{
 		User user = getAppUser();
-		if (!isUserWithBuddies(user))
-		{
-			return null;
-		}
 		List<YonaBuddy> yonaBuddies = user.getBuddies();
 		for (YonaBuddy buddy : yonaBuddies)
 		{
 			if (buddy.getYonaGoals() != null)
 			{
-				return findMatchingGoalFromYonaBuddy(goalHref, buddy);
+				YonaGoal matchingGoal = findMatchingGoalFromYonaBuddy(goalHref, buddy);
+				if (matchingGoal != null)
+				{
+					return matchingGoal;
+				}
 			}
 		}
-		return null; // Dummy return value, to allow use as data load handler
+		return null;
```

Comment:
This is invalid. If we can't find the goal, something is badly wrong and it needs to be flagged right here. Please throw an exception like this:
```
throw new IllegalStateException("Buddy goal not found: " + goalHref.getHref());
```